src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubUploadTest.java

```diff
@@ -71,8 +164,13 @@ private UrlPathPattern setupServerMocking(PlexusConfiguration config, String oas
 
         startMockServer(port);
 
-        UrlPathPattern url = urlPathEqualTo("/apis/" + owner + "/" + api);
+        UrlPathPattern url = stubSaveDefinitionRequest(owner, api, version, isPrivate, oasVersion, format, token);
 
+        return url;
+    }
+
+    private UrlPathPattern stubSaveDefinitionRequest(String owner, String api, String version, String isPrivate, String oasVersion, String format, String token){
+        UrlPathPattern url = urlPathEqualTo("/apis/" + owner + "/" + api);
```

Comment:
Good test coverage, only other thing I can think of is a test covering an incorrect file type? Maybe I missed it

Reply:
So, that's filtered out at the file finder level. I test this in _filesCanBeFetchedFromGivenDirectory_whenNotUsingRegexTest_ . It's not very well documented which i should fix up. Basically the test will return all files that match the valid format. In that directory I have a .gson file which is ignored.